### PR TITLE
Issues/67 add job

### LINKED ISSUE #67: [1st] fix create job logic
# asis
- create job is not working

# tobe
- fix create job logic
- add token to request
- use `react-query`

# related
- #65

### PR DESCRIPTION
**Pull request by azar-barfipour**
## issue numbers to close them automatically

- fixed #67

### CODE REVIEW

**Comment by azar-barfipour** [comment]
Hi. Please review my code. thanks:)

**Comment by azar-barfipour** [comment]
please review

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+//   className: string;
+// }
+
+type Props = {
+  // initialFormData: FormInitialValues;
+  buttonText: string;
+  // onSubmitLogic: (title: string) => void;
+};
+
+const JobForm: FC<Props> = (props) => {
```
Comment (SOunit) [inline src/presentation/components/organisms/CreateJob/EditJobForm.tsx:22]: name should be `EditJobForm`

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  const [jobTitle, setJobTitle] = useState("");
+  const [jobCity, setJobCity] = useState("");
+  const [jobCompany, setJobCompany] = useState("");
+  const [jobPayment, setJobPayment] = useState("");
+  const [jobDesc, setJobDesc] = useState("");
```
Comment (SOunit) [inline src/presentation/components/organisms/CreateJob/EditJobForm.tsx:27]: - jobTitle
- jobCity
- jobCompany
- jobPayment
- jobDesc

let's replace them with `useForm`

Comment (SOunit) [reply src/presentation/components/organisms/CreateJob/EditJobForm.tsx:27]: jobPayment can be string.
backend will make it number.

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  const [skillList, setSkillList] = useState<ISkill[]>([]);
+  // const [checkedSkillList, setCheckedSkillList] = useState<ISkill[]>([]);
+  const { token } = useAppSelector((state) => state.auth);
+
+  const { buttonText } = props;
+
+  // const submitHandler: React.FormEventHandler<HTMLFormElement> = (event) => {
```
Comment (SOunit) [inline src/presentation/components/organisms/CreateJob/EditJobForm.tsx:34]: pls remove all comments

Comment (SOunit) [reply src/presentation/components/organisms/CreateJob/EditJobForm.tsx:34]: or pls put `FIXME` comment if you want to keep comment stay

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    skillListIds.push(skill._id);
+  });
+
+  const formSubmitHandler = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!jobTitle) {
+      return;
+    }
+
+    await axios.post(
```
Comment (SOunit) [inline src/presentation/components/organisms/CreateJob/EditJobForm.tsx:66]: let's use react query package.

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    );
+
+    setJobTitle("");
+  };
+
+  const titleChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
```
Comment (SOunit) [inline src/presentation/components/organisms/CreateJob/EditJobForm.tsx:86]: all input handler can be replaced with `valueChangeHandler` from `useForm`

Comment (SOunit) [reply src/presentation/components/organisms/CreateJob/EditJobForm.tsx:86]: pls check `EditSkillForm.tsx`

File: src/services/jobsHttpClient.adapter.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { API_URL } from "../constants/constants";
+import { IJob } from "./../domain/Job";
+import httpClientAdapter from "./httpClient.adapter";
+
+export const getJobs = (): Promise<IJob[]> => {
```
Comment (SOunit) [inline src/services/jobsHttpClient.adapter.ts:5]: thanks for adding code, but do you need these logic for this issue?
pls add minimum code for your issues.

issue name and PR content should match in working environment, so let's follow same approach here too!

**Review by SOunit — CHANGES_REQUESTED** [review]
thanks! I added some comments. pls check and fix them!

File: src/constants/constants.ts [hunk 7]
```diff
@@ -8,6 +8,7 @@ export const TOKEN_KEY = "JOB_SEARCH_APP_TOKEN_KEY";
 export const USER_ID_KEY = "JOB_SEARCH_APP_USER_ID_KEY";
 
 export const REACT_QUERY_KEY_SKILLS = "skills";
+export const REACT_QUERY_KEY_JOBS = "skills";
```
Comment (SOunit) [inline src/constants/constants.ts:11]: value should be "jobs"?

**Review by SOunit — CHANGES_REQUESTED** [review]

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx (hunk 4 above)
Comment (SOunit) [reply src/presentation/components/organisms/CreateJob/EditJobForm.tsx:66]: but sorry, as a first step, it's ok using axios :)

**Review by SOunit — CHANGES_REQUESTED** [review]

File: src/presentation/components/organisms/CreateJob/EditJobForm.tsx (hunk 1 above)
Comment (azar-barfipour) [reply src/presentation/components/organisms/CreateJob/EditJobForm.tsx:22]: done

**Review by azar-barfipour — COMMENTED** [review]
lgtm

**Review by SOunit — APPROVED** [review]
lgtm

**Pull request merged**